<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/quotes.ts
  - ../../tests/rules/quotes/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/template/quotes`

Specify whether to use double quotes or single quotes in template attributes

- Type: layout
- üîß Supports autofix (`--fix`)

<br>

## Rule Options

The rule accepts an options object with the following properties:

```ts
interface Options {
  /**
   * Default: `"double"`
   */
  quotesType?: "double" | "single";
}

```

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>‚ùå - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/quotes": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<div class='my-class'></div>
           ~~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/quotes": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<my-component [input]='myInput'></my-component>
                      ~~~~~~~~~
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/quotes": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<my-component (output)='myFunc()'></my-component>
                       ~~~~~~~~~~
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/template/quotes": [
      "error",
      {
        "quotesType": "single"
      }
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<div class="my-class"></div>
           ~~~~~~~~~~
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/template/quotes": [
      "error",
      {
        "quotesType": "single"
      }
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<my-component [input]="myInput"></my-component>
                      ~~~~~~~~~
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/template/quotes": [
      "error",
      {
        "quotesType": "single"
      }
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```html
<my-component (output)="myFunc()"></my-component>
                       ~~~~~~~~~~
```

</details>

<br>

---

<br>

<details>
<summary>‚úÖ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/quotes": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<div class="my-class"></div>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/quotes": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<my-component [input]="myInput"></my-component>
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/template/quotes": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<my-component (output)="myFunc()"></my-component>
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/template/quotes": [
      "error",
      {
        "quotesType": "single"
      }
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<div class='my-class'></div>
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/template/quotes": [
      "error",
      {
        "quotesType": "single"
      }
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<my-component [input]='myInput'></my-component>
```

<br>

---

<br>

#### Custom Config

```json
{
  "rules": {
    "@angular-eslint/template/quotes": [
      "error",
      {
        "quotesType": "single"
      }
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```html
<my-component (output)='myFunc()'></my-component>
```

</details>

<br>
